app/lib/actions/pulp3/capsule_content/generate_metadata.rb

```diff
@@ -6,6 +6,7 @@ class GenerateMetadata < Pulp3::AbstractAsyncTask
         def plan(repository, smart_proxy, options = {})
           options[:contents_changed] = (options && options.key?(:contents_changed)) ? options[:contents_changed] : true
           sequence do
+            plan_action(RefreshContentGuard, smart_proxy) unless repository.unprotected
```

Comment:
we might need to run this regardless of repository.unprotected in case the value for that changed

Reply:
right, but what if you 'unset' the unprotected flag, in which case 'unprotected' will be false and this won't ever get called, resulting in the repo still being protected, right?

Reply:
oh, right, the actual association to the content guard is done as part of refreshrepos